### DISCUSSION TITLE
[Feature] Face group search option on public shared link

### DISCUSSION THREAD

**Discussion by deven-softices** [immich-app/immich#12488] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

There should be a face group search/group by face option on the public shared album to easily find relevant photos group by face.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by histefanhere** [comment]
I thought of this idea the other day, glad it's already been requested here! I often share albums with hundreds of photos to 20+ people (which are _horrible_ to look through for them because the norm now is to use your phone for everything but the UI isn't mobile optimized due to #4798...), but to find photos of themselves they need to look through all the photos which can get daunting. A solution to this would be a banner of people and their faces at the top of the share page, possibly people who have been named in Immich and appear in any of the photos within the album. Clicking on any of the faces would filter images to only show photos that include said person.

A rough plan created by me for implementing this feature:
1. When a shared album is viewed, collate a list of named people in all the photos in the album.
2. Show this list of people on the page as a row of faces.
3. When clicked on, filter the images shown by images that feature the selected person.
4. Show the person's face separately, with an `x`, or a `clear filter` button etc. for clearing the filter that when clicked shows all photos in the album.

I hope we get to see this in Immich one day soon!

**Comment by DrLabel** [comment]
I was looking for this, My wife takes photos for a hobby of sporting events where our daughter participates, then puts everything in the immich creates an album and shares it with other people, it was good that other people could upload their face or select the detected faces to make it easier to see their photos
